tests/contrib/openstack/test_os_contexts.py

```diff
@@ -50,10 +50,16 @@ class FakeRelation(object):
         passwd = self.relation_get('password', rid='mysql:0', unit='mysql/0')
     '''
 
-    def __init__(self, relation_data):
+    def __init__(self, relation_data={}, app_data={}):
```

Comment:
This is problematic as dictionaries have internal mutability, and are only initialised on the first run of the function; therefore, the function will memoize the `relation_data` and `app_data` between calls.  It's better to do:

```python
def __init__(self, relation_data=None, app_data=None):
    if relation_data is None:
        relation_data = {}
    if app_data is None:
        app_data = {}
    ...
```